packages/react-native/src/message-list/message-list.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { FC, useEffect, useRef, useCallback, useState } from "react";
+import type { NativeSyntheticEvent, NativeScrollEvent } from "react-native";
+import {
+  Animated,
+  FlatList,
+  Image,
+  ListRenderItem,
+  Platform,
+  Pressable,
+  Text,
+  View,
+} from "react-native";
+import {
+  CommonMessageListProps,
+  MessageEnvelope,
+  getNameInitials,
+  getPredefinedColor,
+  isFilePayload,
+  useMessageListCore,
+} from "@pubnub/common-chat-components";
+import createDefaultStyle, { MessageListStyle } from "./message-list.style";
+import { useStyle, useRotation } from "../helpers";
+import SpinnerIcon from "../icons/spinner.png";
+
+export type MessageListProps = CommonMessageListProps & {
+  /** Callback run on a list scroll. */
+  onScroll?: (event: NativeSyntheticEvent<NativeScrollEvent>) => void;
+  /** Options to provide custom StyleSheet for the component. It will be merged with the default styles. */
+  style?: MessageListStyle;
+};
+
+/**
+ * Fetches historical messages using the scrolling pagination pattern and subscribes to the current
+ * channel to stay up to date with new messages.
+ *
+ * It also displays data in an interactive list, including
+ * user names, avatars, the time when a message was sent, and attachments (links, images) and allows to react to
+ * messages with emojis and to show those reactions immediately.
+ */
+export const MessageList: FC<MessageListProps> = (props: MessageListProps) => {
+  const {
+    fetchHistory,
+    getTime,
+    getUser,
+    isOwnMessage,
+    messages,
+    paginationEnd,
+    prevMessages,
+    setScrolledBottom,
+    scrolledBottom,
+    setUnreadMessages,
+    theme,
+    unreadMessages,
+  } = useMessageListCore(props);
+
+  const style = useStyle<MessageListStyle>({
+    theme,
+    createDefaultStyle,
+    customStyle: props.style,
+  });
+
+  const [reverseMessages, setReverseMessages] = useState([...messages].reverse());
+  const [spinnerShown, setSpinnerShown] = useState(false);
+  const shouldShownSpinner = props.fetchMessages && !paginationEnd;
+  const isAndroid = Platform.OS === "android";
+
+  const rotate = useRotation(shouldShownSpinner && spinnerShown);
+  const listRef = useRef<FlatList>(null);
+
+  /**
+   * Commands
+   */
+  const scrollToBottom = useCallback(() => {
+    setScrolledBottom(true);
+    listRef.current.scrollToOffset({ offset: 0 });
+  }, [setScrolledBottom]);
+
+  /**
+   * Handlers
+   */
+  const handleScroll = useCallback(
```

Comment:
Are you sure you don't wanna debounce scroll events?

Reply:
You should probably do it in JS cause native solution works only for iOS https://reactnative.dev/docs/scrollview#scrolleventthrottle-ios

Reply:
Fixed